### PR TITLE
New InfraEng updates

### PR DESCRIPTION
**Pull request by cabutlermit**
### Why are these changes being introduced

New documentation for the new AWS Organization and the new Terraform 1.0 repositories. Move old AWS and Terraform documentation to "legacy" status (since they are still valid, just not for any new resources).

### How does this address that need

* rename the aws and terraform `.rst` files to include "legacy"
* create new `aws-2.0.md` and `terraform-1.0.md` files
* the new aws-2.0 page includes 
    * information about our new AWS Organization
    * information about the new AWS SSO based sign-in
    * links to more detailed documentation in the InfraEng Confluence wiki
 * the new terraform-1.0 page includes
    * information about our new Terraform repository model
    * information about our Git flow (vs GitHub flow) approach to our Terraform repos
    * information about our automated management of repos
    * information about our Terraform Cloud Organization
    * links to more detailed documentation in the InfraEng Confluence wiki

### Other Notes

I'm looking for feedback from InfraEng that the information being added to our DevDocs accurately represents our new approach to AWS, Terraform, and infrastructure.

I'm looking for feedback from EngX that the information about access to AWS and the notes about our approach to managing branches in our repositories makes sense.

### CODE REVIEW

File: basics/github.md [hunk 1]
```diff
@@ -11,13 +11,16 @@ In the admin settings for repos on GitHub, the recommended settings are:
...
 
 Ideally, feature branches should encompass small, targeted changes. Keeping your work focused will mean fewer merge conflicts. When building a new code base this may not always be possible, but what you want to try and avoid are long running branches. By the time you want to merge, the distance between your feature branch and main will usually mean a lot of conflict resolution in your future.
 
+There is an exception to this for Infrastructure code. For the Terraform code, InfraEng uses a Simplified Git Flow model (which could also be seen as GitHub Flow on steroids). In this case, each deployment environment (prod & stage) is linked to a long-running branch in the repository. As in GitHub Flow, feature branches should encompass small, targeted changes. Then are always branched from and merged to the `stage` branch. And the only merges to the `main` branch come from approved merges on the `stage` branch. More details on this model can be found at [Simplified Git Flow](https://medium.com/goodtogoat/simplified-git-flow-5dc37ba76ea8) and at [Simplified Git Flow for InfraEng](https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/2920480769/Simplified+Git+Flow+For+InfraEng).
```
Comment (tim-rix) [inline basics/github.md]: `Then are always branched from and merged`

Should this be "They"?

Otherwise - everything looks good

**Review by tim-rix — APPROVED** [review]

File: basics/github.md (hunk 1 above)
Comment (cabutlermit) [reply basics/github.md]: fixed.

File: basics/github.md [hunk 2]
```diff
@@ -157,7 +160,7 @@ git push
 
 **Get thumbs**: once each reviewer is done with their review and happy with the code that will be merged, they should approve the PR and give a thumbs up  :+1:
 
-**Squash commits (optional)**: before merging, squash all your work into logical commits - or one single commit - that summarizes the changes.
+**Squash commits (optional)**: before merging, squash all your work into logical commits - or one single commit - that summarizes the changes. **Note**: try to avoid using squash commits to do the squashing!
```
Comment (JPrevost) [inline basics/github.md]: I feel like we have some confusing language going on. The difference between "squashing commits" and "squash commits" is unfortunate.

Maybe the Note could be something like "do not use GitHub's option to use squash commits to do the squashing"

File: basics/github.md [hunk 3]
```diff
@@ -11,13 +11,16 @@ In the admin settings for repos on GitHub, the recommended settings are:
     - require status checks to pass before merging
     - require that branches are up to date
 * Allow merge commits (`Options > Merge button`)
+* **Optional** Disable squash commits (`Options > Merge button`)
```
Comment (JPrevost) [inline basics/github.md]: I'd suggest we make it either `Recommended` or a new `Requirement` to disable squash commits. EngX never uses this feature and had it disabled explicitly in the repos we use most often and have remembered to change this.

**Review by JPrevost — APPROVED** [review]
I have left some comments, but I don't consider them blocking. You can merge as-is or make some small changes as noted. Either way thanks for updating docs as processes change!

File: basics/github.md (hunk 2 above)
Comment (cabutlermit) [reply basics/github.md]: I totally agree. I will re-word this section a bit.

File: basics/github.md (hunk 3 above)
Comment (cabutlermit) [reply basics/github.md]: I will shift this to "Requirement" for our Terraform repos. GitHub squash commits on a multi-long-running-branch repo is a recipe for disaster.

**Pull request merged**